contrib/inventory/scaleway.py

```diff
@@ -0,0 +1,223 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import ConfigParser
+import copy
+import os
+import requests
+import sys
+import time
+import traceback
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+EMPTY_GROUP = {
+    'children': [],
+    'hosts': []
+}
+
+
+class ScalewayAPI:
+    REGIONS = ['par1', 'ams1']
+
+    def __init__(self, auth_token, region):
+        self.session = requests.session()
+        self.session.headers.update({
+            'User-Agent': 'Ansible Python/%s' % (sys.version.split(' ')[0])
+        })
+        self.session.headers.update({
+            'X-Auth-Token': auth_token.encode('latin1')
+        })
+        self.base_url = 'https://cp-%s.scaleway.com' % (region)
+
+    def servers(self):
+        raw = self.session.get('/'.join([self.base_url, 'servers']))
+
+        try:
+            response = raw.json()
+            return self.get_resource('servers', response, raw)
+        except ValueError:
+            return []
+
+    def get_resource(self, resource, response, raw):
+        raw.raise_for_status()
+
+        if resource in response:
+            return response[resource]
+        else:
+            raise ValueError(
+                "Resource %s not found in Scaleway API response" % (resource))
+
+
+def env_or_param(env_key, param=None, fallback=None):
+    env_value = os.environ.get(env_key)
+
+    if (param, env_value) == (None, None):
+        return fallback
+    elif env_value is not None:
+        return env_value
+    else:
+        return param
+
+
+def save_cache(data, config):
+    ''' saves item to cache '''
+    dpath = config.get('cache', 'cache_dir')
+    try:
+        cache = open('/'.join([dpath, 'scaleway_ansible_inventory.json']), 'w')
+        cache.write(json.dumps(data))
+        cache.close()
+    except IOError as e:
+        pass  # not really sure what to do here
+
+
+def get_cache(cache_item, config):
+    ''' returns cached item  '''
+    dpath = config.get('cache', 'cache_dir')
+    inv = {}
+    try:
+        cache = open('/'.join([dpath, 'scaleway_ansible_inventory.json']), 'r')
+        inv = cache.read()
+        cache.close()
+    except IOError as e:
+        pass  # not really sure what to do here
+
+    return inv
+
+
+def cache_available(config):
+    ''' checks if we have a 'fresh' cache available for item requested '''
+
+    if config.has_option('cache', 'cache_dir'):
+        dpath = config.get('cache', 'cache_dir')
+
+        try:
+            existing = os.stat(
+                '/'.join([dpath, 'scaleway_ansible_inventory.json']))
+        except:
```

Comment:
I tend to dislike unlimited except, since this can hide problem that shouldn't be hidden. Can this be limited somehow to just catch a few exception ?

Reply:
You are right. I have fixed that now.